src/mapgen_carpathian.cpp

```diff
@@ -0,0 +1,451 @@
+	settings->getNoiseParams("mgcarpathian_np_filler_depth",  np_filler_depth);
+	settings->getNoiseParams("mgcarpathian_np_height1",       np_height1);
+	settings->getNoiseParams("mgcarpathian_np_height2",       np_height2);
+	settings->getNoiseParams("mgcarpathian_np_height3",       np_height3);
+	settings->getNoiseParams("mgcarpathian_np_height4",       np_height4);
+	settings->getNoiseParams("mgcarpathian_np_hills_terrain", np_hills_terrain);
+	settings->getNoiseParams("mgcarpathian_np_ridge_terrain", np_ridge_terrain);
+	settings->getNoiseParams("mgcarpathian_np_step_terrain",  np_step_terrain);
+	settings->getNoiseParams("mgcarpathian_np_hills",         np_hills);
+	settings->getNoiseParams("mgcarpathian_np_ridge_mnt",     np_ridge_mnt);
+	settings->getNoiseParams("mgcarpathian_np_step_mnt",      np_step_mnt);
+	settings->getNoiseParams("mgcarpathian_np_mnt_var",       np_mnt_var);
+	settings->getNoiseParams("mgcarpathian_np_cave1",         np_cave1);
+	settings->getNoiseParams("mgcarpathian_np_cave2",         np_cave2);
+	settings->getNoiseParams("mgcarpathian_np_cavern",        np_cavern);
+}
+
+
+void MapgenCarpathianParams::writeParams(Settings *settings) const
+{
+	settings->setFlagStr("mgcarpathian_spflags", spflags, flagdesc_mapgen_carpathian, U32_MAX);
+	settings->setFloat("mgcarpathian_cave_width",       cave_width);
+	settings->setS16("mgcarpathian_cavern_limit",       cavern_limit);
+	settings->setS16("mgcarpathian_cavern_taper",       cavern_taper);
+	settings->setFloat("mgcarpathian_cavern_threshold", cavern_threshold);
+
+	settings->setNoiseParams("mgcarpathian_np_base",          np_base);
+	settings->setNoiseParams("mgcarpathian_np_filler_depth",  np_filler_depth);
+	settings->setNoiseParams("mgcarpathian_np_height1",       np_height1);
+	settings->setNoiseParams("mgcarpathian_np_height2",       np_height2);
+	settings->setNoiseParams("mgcarpathian_np_height3",       np_height3);
+	settings->setNoiseParams("mgcarpathian_np_height4",       np_height4);
+	settings->setNoiseParams("mgcarpathian_np_hills_terrain", np_hills_terrain);
+	settings->setNoiseParams("mgcarpathian_np_ridge_terrain", np_ridge_terrain);
+	settings->setNoiseParams("mgcarpathian_np_step_terrain",  np_step_terrain);
+	settings->setNoiseParams("mgcarpathian_np_hills",         np_hills);
+	settings->setNoiseParams("mgcarpathian_np_ridge_mnt",     np_ridge_mnt);
+	settings->setNoiseParams("mgcarpathian_np_step_mnt",      np_step_mnt);
+	settings->setNoiseParams("mgcarpathian_np_mnt_var",       np_mnt_var);
+	settings->setNoiseParams("mgcarpathian_np_cave1",         np_cave1);
+	settings->setNoiseParams("mgcarpathian_np_cave2",         np_cave2);
+	settings->setNoiseParams("mgcarpathian_np_cavern",        np_cavern);
+}
+
+
+///////////////////////////////////////////////////////////////////////////////
+
+
+// Lerp function
+float MapgenCarpathian::getLerp(float noise1, float noise2, float mod)
+{
+	return (1.f - mod) * noise1 + mod * noise2;
```

Comment:
mnt_var noise is:
```
+	np_mnt_var       = NoiseParams(0,  1,  v3f(499,  499,  499),  2490,  5, 0.6,  2.0);
```
in case you did not know, with scale 1 and 5 octaves persistence 0.6 the value will actually vary between roughly -2 and 2.
Specifically the variation will be 1 + 0.6 + 0.6^2 + 0.6^3 + 0.6^4.
So if you want you might prefer to limit `mod` to -1 to 1.